Build-Toolkit-Components.ps1

```diff
@@ -0,0 +1,196 @@
+<#
+.SYNOPSIS
+    Builds toolkit components with specified parameters. Primarily used by maintainers for local testing.
+
+.DESCRIPTION
+    This script streamlines building and packing Community Toolkit components with the specified parameters. It allows you to specify the MultiTarget TFM(s) to include or exclude, the WinUI major version to use, the components to build, whether to build samples or source, optional packing when provided with a NupkgOutput, and more. The components can be built in Release configuration and individual (per-component) binlogs can be generated by passing -bl.
+
+.PARAMETER MultiTargets
+    Specifies the MultiTarget TFM(s) to include for building the components. The default value is 'all'.
+
+.PARAMETER ExcludeMultiTargets
+    Specifies the MultiTarget TFM(s) to exclude for building the components. The default value excludes targets that require additional tooling or workloads to build. Run uno-check to install the required workloads.
+
+.PARAMETER DateForVersion
+    Specifies the date for versioning in 'YYMMDD' format. The default value is the current date.
+
+.PARAMETER PreviewVersion
+    Specifies the preview version to use if packaging is enabled. Appended with a dash after the version number (formatted Version-PreviewVersion). This parameter is required when NupkgOutput is supplied.
```

Comment:
@Arlodotexe missed this, this should be optional, this is blocking our release now as we won't have a preview version postfix.